static/src/javascripts/projects/common/modules/discussion/api.ts

```diff
@@ -40,52 +58,69 @@ const defaultInitParams: RequestInit = {
 	},
 };
 
-export const send = (
+export const send = <T = CommentResponse>(
 	endpoint: string,
-	method: string,
+	method: 'GET' | 'POST',
 	data?: Comment | AbuseReport,
-): Promise<CommentResponse> => {
-	if (config.get('switches.enableDiscussionSwitch')) {
-		const apiUrl = config.get<string>(
-			'page.discussionApiUrl',
-			'/DISCUSSION_API_URL_NOT_FOUND',
-		);
-		const url = apiUrl + endpoint;
-
-		// https://github.com/guardian/discussion-rendering/blob/1e8a7c7fa0b6a4273497111f0dab30f479a107bf/src/lib/api.tsx#L140
-		if (method === 'POST') {
-			const body = new URLSearchParams();
-
-			if (data) {
-				if ('body' in data) {
-					body.append('body', data.body);
-				}
-				if ('categoryId' in data) {
-					body.append('categoryId', data.categoryId.toString());
-					data.email && body.append('email', data.email.toString());
-					data.reason && body.append('reason', data.reason);
+): Promise<T> =>
+	Promise.resolve(config.get('switches.enableDiscussionSwitch'))
+		.then((isDiscussionEnabled) =>
+			isDiscussionEnabled
+				? Promise.resolve()
+				: Promise.reject('switches.enableDiscussionSwitch is off'),
+		)
```

Comment:
maybe this pattern is actually harder to read? I quite like it but would like to know what other people think!